components/ActionBar/Article.js

```diff
@@ -1,133 +1,133 @@
-import React, { Component, Fragment } from 'react'
+import React, { Component, Fragment, useState, useEffect } from 'react'
 import ActionBar from './'
 import DiscussionIconLink from '../Discussion/IconLink'
 import { getDiscussionIconLinkProps } from './utils'
 
 import { fontStyles } from '@project-r/styleguide'
 import ShareButtons from './ShareButtons'
+import UserProgress from './UserProgress'
 
-class ArticleActionBar extends Component {
-  constructor(props) {
-    super(props)
+const ArticleActionBar = ({
+  animate,
+  title,
+  tweet,
+  emailBody,
+  emailAttachUrl,
+  template,
+  path,
+  linkedDiscussion,
+  ownDiscussion,
+  documentId,
+  repoId,
+  isEditor,
+  dossierUrl,
+  estimatedReadingMinutes,
+  estimatedConsumptionMinutes,
+  onAudioClick,
+  onGalleryClick,
+  onPdfClick,
+  pdfUrl,
+  showBookmark,
+  t,
+  url,
+  inNativeApp,
+  inIOS,
+  showShare = true,
+  grandSharing,
+  fontSize,
+  userBookmark,
+  userProgress,
+  showSubscribe,
+  subscription,
+  isDiscussion
+}) => {
+  const {
+    discussionId,
+    discussionPath,
+    discussionQuery,
+    discussionCount,
+    isDiscussionPage
+  } = getDiscussionIconLinkProps(
+    linkedDiscussion,
+    ownDiscussion,
+    template,
+    path
+  )
+  const emailSubject = t('article/share/emailSubject', { title })
+  const [alive, setAlive] = useState(false)
+  useEffect(() => {
+    setAlive(true)
+  })
```

Comment:
```suggestion
  }, [])
```
since we are replacing `componentDidMount`

Reply:
thx! 🙏🏼